server/graphql/v1/mutations/notifications.js

```diff
@@ -0,0 +1,88 @@
+import Promise from 'bluebird';
+import { values } from 'lodash';
+import models, { Op } from '../../../models';
+import activities from '../../../constants/activities';
+import { channels } from '../../../constants';
+import { Forbidden } from '../../errors';
+
+const NotificationPermissionError = new Forbidden({
+  message: "This notification does not exist or you don't have the permission to edit it.",
+});
+
+export async function editNotifications(args, remoteUser) {
+  if (!remoteUser) {
+    throw NotificationPermissionError;
+  }
+
+  if (!args.notifications) return Promise.resolve();
+  let newNotifications = [];
+
+  return models.Notification.findAll({ where: { CollectiveId: args.id, UserId: null } })
+    .then(oldNotifications => {
+      const diff = oldNotifications
+        .filter(
+          x1 =>
+            !args.notifications.some(x2 => {
+              return x2.channel === x1.channel && x2.type === x1.type && x2.webhookUrl === x1.webhookUrl;
+            }),
+        )
+        .map(x => x.id);
+
+      newNotifications = args.notifications.filter(
+        x1 =>
+          !oldNotifications.some(x2 => {
+            return x2.channel === x1.channel && x2.type === x1.type && x2.webhookUrl === x1.webhookUrl;
+          }),
+      );
+
+      return models.Notification.destroy({ where: { id: { [Op.in]: diff } } });
+    })
+    .then(() => {
+      return Promise.map(newNotifications, notification => {
+        if (!(values(activities).includes(notification.type) && values(channels).includes(notification.channel)))
+          return;
+
+        notification.CollectiveId = args.id;
+        return models.Notification.create(notification);
+      });
+    });
+}
+
+/**
+ * Creates a new notification
+ */
+export async function createNotification(args, remoteUser) {
+  if (!remoteUser) {
+    throw NotificationPermissionError;
+  }
+
+  const { CollectiveId, webhookUrl, channel, type } = args.notification;
+
+  return await models.Notification.create({
+    UserId: remoteUser.id,
+    CollectiveId,
+    webhookUrl,
+    channel,
+    type,
+  });
+}
+
+/**
+ * Creates a Webhook subscription for a collective given a collective slug.
+ */
+export async function createWebhook(args, remoteUser) {
+  const collective = models.Collective.find({ where: { slug: args.collectiveSlug } });
+
+  args.notification.channel = channels.WEBHOOK;
+  args.notification.CollectiveId = collective.id;
+
+  return createNotification(args, remoteUser);
+}
+
+export async function deleteNotification(args, remoteUser) {
+  if (!remoteUser) {
```

Comment:
We also need to check that `remoteUser` is admin of collective `CollectiveId`. See https://github.com/opencollective/opencollective-api/blob/32f473bdbd17056bf8dbef7e6e6c392c32fa8747/server/graphql/v1/mutations/collectives.js#L691 for an example

Reply:
Since non-admins will be able to register webhooks, they should also be allowed to delete webhooks they register.

Reply:
Yes, the check below (`remoteUser.id === notification.UserId || remoteUser.isAdmin(notification.CollectiveId)`) looks good to me